stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -1082,6 +1086,17 @@ internal constructor(
         disconnect(true).execute()
     }
 
+    /**
+     * Cancel all jobs on the ChatClient Scope.
+     * This method postpone a new coroutine to cancel all children jobs.
+     */
+    private fun postponeCancelScope() {
+        scope.launch {
+            delay(DELAY_TIME_TO_CANCEL_CHILDREN)
```

Comment:
Any reason why we postpone here?

Reply:
Just saw the previous comment - if we could do it in a non.delay manner, it would be great, but I guess this is fine for now.

Reply:
As I commented on the previous comment, it is needed to be able to pass our tests, because when `connectUser()` fails, `disconnect()` method is called (To keep a clean state in our `ChatClient`), but because we are using the same coroutineScope and it is canceled, our tests fail

Reply:
It should be fine to keep it for now. Later we can investigate if we can somehow get rid of it